cql2pgjson/src/test/java/org/z3950/zing/cql/cql2pgjson/ForeignKeyGenerationIT.java

```diff
@@ -75,13 +76,16 @@ public void uuidConstant() throws Exception {
     assertThat(cql("tableb.name == 33333333-3333-3333-3333-333333333333"), is(empty()));
   }
   @Test
-  public void ForeignKeySearchWithInjection1() throws FieldException, QueryValidationException, ServerChoiceIndexesException {
-
+  public void ForeignKeySearchWithInjection1() {
+    //check to see if we can execute some arbitrary sql
     assertThat(cql("tableb.prefix == \"x0')));((('DROP tableb\""), containsInAnyOrder("test3") );
+    //then check to see if the drop table worked by checking to see if there is anything there
+    assertThat(cql("tableb.prefix == \"x0')));((('DROP tableb\"").size() > 0, is(true));
```

Comment:
If there is some injection resulting in executing both the select statement and the drop statement then the select statement returns 0 results and the drop statement returns 0 results.

A more easy way to check for injection is one (= unbalanced) single quote that triggers an SQL parse error that will raise a database exception in java:

    assertThat(cql("tableb.prefix == '"), is(empty()));

Reply:
There is no need to run the same CQL query twice.